src/CLICommand/action.ts

```diff
@@ -0,0 +1,131 @@
+import { ActionReturnType, AsyncActionCallback, ParsedParameters } from './cli';
+import { ERROR_EXIT_CODE } from './error_codes';
+
+/**
+ * A wrapper for the command action callback
+ */
+export class CLIAction {
+	private _promiseInstance?: Promise<ActionReturnType>;
+	private _parsedParameters: ParsedParameters = {};
+	private readonly _awaiterInstances: ((value?: ParsedParameters, error?: Error) => void)[] = [];
+	private static _runningAction?: CLIAction;
+
+	/**
+	 * Create an instance of CLIAction
+	 * @param {AsyncActionCallback} actionCallback the handler method of the action
+	 */
+	constructor(
+		private readonly actionCallback: AsyncActionCallback = async (opts) => {
```

Comment:
What are the technical difficulties of achieving type safety here? When we define an action, shouldn't these be our parameters for the action?

Reply:
It was the initial workaround approach for the Promise catching the Mocha expect() assertions
7ab596fb75ae61696dd218acf2262754c7441660

Reply:
That default value was removed